hazelcast/config.py

```diff
@@ -439,3 +447,81 @@ def eviction_sampling_pool_size(self, eviction_sampling_pool_size):
+    Client properties can be set by
+    * Programmatic Configuration (config.set_property)
+    * Environment variables
+    """
+    def __init__(self, name, default_value=None, time_unit=None):
+        self.name = name
+        self.default_value = default_value
+        self.time_unit = time_unit
+
+
+class ClientProperties(object):
+    HEARTBEAT_INTERVAL = ClientProperty("hazelcast.client.heartbeat.interval", 5000, TimeUnit.MILLISECOND)
+    """
+    Time interval between the heartbeats sent by the client to the nodes.
+    """
+
+    HEARTBEAT_TIMEOUT = ClientProperty("hazelcast.client.heartbeat.timeout", 60000, TimeUnit.MILLISECOND)
+    """
+    Client sends heartbeat messages to the members and this is the timeout for this sending operations.
+    If there is not any message passing between the client and member within the given time via this property
+    in milliseconds, the connection will be closed.
+    """
+
+    INVOCATION_TIMEOUT_SECONDS = ClientProperty("hazelcast.client.invocation.timeout.seconds", 120, TimeUnit.SECOND)
+    """
+    When an invocation gets an exception because :
+    * Member throws an exception.
+    * Connection between the client and member is closed.
+    * Client's heartbeat requests are timed out.
+    Time passed since invocation started is compared with this property.
+    If the time is already passed, then the exception is delegated to the user. If not, the invocation is retried.
+    Note that, if invocation gets no exception and it is a long running one, then it will not get any exception,
+    no matter how small this timeout is set.
+    """
+
+    INVOCATION_RETRY_PAUSE_MILLIS = ClientProperty("hazelcast.client.invocation.retry.pause.millis", 1000,
+                                                   TimeUnit.MILLISECOND)
+    """
+    Pause time between each retry cycle of an invocation in milliseconds.
+    """
+
+    def __init__(self, properties):
+        self._properties = properties
+
+    def get(self, property):
+        """
+        Gets the value of the given property. First checks client config properties, then environment variables
+        and lastly fall backs to the default value of the property.
+        :param property: (:class:`~hazelcast.config.ClientProperty`), Property to get value from
+        :return: Returns the value of the given property
+        """
+        return self._properties.get(property.name) or os.getenv(property.name) or property.default_value
+
+    def get_seconds(self, property):
+        """
+        Gets the value of the given property in seconds. If the value of the given property is not a number,
+        throws TypeError.
+        :param property: (:class:`~hazelcast.config.ClientProperty`), Property to get seconds from
+        :return: (float), Value of the given property in seconds
+        """
+        return TimeUnit.to_seconds(self.get(property), property.time_unit)
+
+    def get_seconds_positive(self, property):
```

Comment:
Could a name like get_seconds_positive_or_default be better ? It seems long but it describes what the function does better.